src/test/java/org/assertj/core/internal/objectarrays/ObjectArrays_assertHasExactlyElementsOfTypes_Test.java

```diff
@@ -0,0 +1,96 @@
+package org.assertj.core.internal.objectarrays;
+
+import java.util.LinkedList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.error.ShouldContainExactly.elementsDifferAtIndex;
+import static org.assertj.core.error.ShouldContainExactly.shouldContainExactly;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.util.Arrays.asList;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.assertj.core.util.Lists.newArrayList;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ObjectArrays;
+import org.assertj.core.internal.ObjectArraysBaseTest;
+import org.assertj.core.internal.StandardComparisonStrategy;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link ObjectArrays#assertHasExactlyElementsOfTypes(AssertionInfo, Object[], Class...)}</code>.
+ */
+class ObjectArrays_assertHasExactlyElementsOfTypes_Test extends ObjectArraysBaseTest
```

Comment:
read https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md and adjust the tests accordingly, basically use GIVEN WHEN THEN for each tests and `expectAssertionError` for failure testing